testng-core/src/main/java/org/testng/internal/invokers/TestInvoker.java

```diff
@@ -726,6 +726,9 @@ private ITestResult invokeMethod(
       StatusHolder holder =
           considerExceptions(
               arguments.getTestMethod(), testResult, expectedExceptionClasses, failureContext);
+      if (holder.status != testResult.getStatus()) {
```

Comment:
No need to check, you can always set the value. Please add a comment that explains **why** we override the status here.

Reply:
Fixed